### PR TITLE
feat(cli): get latest dependencies versions from npm

### PR DESCRIPTION
**Pull request by eladav**
* Read package file and get versions for required dependencies
* Lowercase package name
solves #71

### CODE REVIEW

**Comment by eladav** [comment]
Couldn't find any way to get versions for multiple packages with one request... I followed the instructions here for the metadata request:
https://github.com/npm/registry/blob/master/docs/responses/package-metadata.md

File: client/cli/lib/init.ts [hunk 1]
```diff
@@ -9,7 +8,7 @@ const dirname = process
   .pop() as string;
 
 export default () =>
-  execa('npm', ['run', 'plop', 'component', process.cwd(), uppercamelcase(dirname)], {
+  execa('npm', ['run', 'plop', 'component', process.cwd(), dirname], {
```
Comment (EladBezalel) [inline client/cli/lib/init.ts]: Should we do lower tho?

File: client/cli/plop/utils.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import fetch from 'node-fetch';
+
+const getDependencyMetadata = async (name: string) =>
+  fetch(`https://registry.npmjs.org/${name}`, {
```
Comment (EladBezalel) [inline client/cli/plop/utils.ts]: Remove async

File: client/cli/plop/utils.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    headers: { Accept: 'application/vnd.npm.install-v1+json; q=1.0, application/json; q=0.8, */*' }
+  }).then(r => r.json());
+
+export const getVersionsForDependencies = async (deps: string[]) => {
+  return (await Promise.all(deps.map(d => getDependencyMetadata(d)))).reduce(
```
Comment (EladBezalel) [inline client/cli/plop/utils.ts]: One line

File: client/cli/plop/utils.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    (soFar: Record<string, string>, current: any) => ({
```
Comment (EladBezalel) [inline client/cli/plop/utils.ts]: Why any?

**Comment by eladav** [comment]
matter of seconds on WiFi...

File: client/cli/lib/init.ts (hunk 1 above)
Comment (eladav) [reply client/cli/lib/init.ts]: not sure... at this point it just shows the name of the folder... I'm running lower in the name property in package.json file... I'm considering whether or not we should auto kebab-case it...?

Comment (EladBezalel) [reply client/cli/lib/init.ts]: Nah I think it's fine to leave it be

File: client/cli/plop/component/index.ts [hunk 5]
```diff
@@ -59,7 +60,15 @@ export default (plop: Plop) => {
       {
         type: 'add',
         path: '{{path}}/package.json',
-        templateFile: 'plop/component/{{framework}}/package.hbs'
+        templateFile: 'plop/component/{{framework}}/package.hbs',
+        data: async () => {
+          const { dependencies, devDependencies, peerDependencies } = JSON.parse(
+            plop.renderString(readFileSync(`./plop/component/${data.framework}/package.hbs`, 'utf-8'), data)
+          );
+          return await getVersionsForDependencies(
```
Comment (EladBezalel) [inline client/cli/plop/component/index.ts]: I don't think we need async await here

File: client/cli/plop/utils.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+import { PackageMetadata } from './typings';
+
+const getDependencyMetadata = (name: string) =>
+  fetch(`https://registry.npmjs.org/${name}`, {
+    headers: { Accept: 'application/vnd.npm.install-v1+json; q=1.0, application/json; q=0.8, */*' }
+  }).then(r => r.json());
+
+export const getVersionsForDependencies = async (deps: string[]) =>
+  (await Promise.all(deps.map(d => getDependencyMetadata(d)))).reduce(
```
Comment (EladBezalel) [inline client/cli/plop/utils.ts]: Also here do we have to use async await?
Returning promise is enough imo

Comment (eladav) [reply client/cli/plop/utils.ts]: So I can ‘reduce’ on the promise.all result to a name:version object

**Pull request merged**